### PR TITLE
feat(state-representation): add utilities for CartesianStateVariable

### PR DESCRIPTION
**Pull request by bpapaspyros**
## Description

- #195 

This PR solves the issue by making `get_state_variable` and `set_state_variable` public, and also adding two support functions to convert `CartesianStateVariable`s from/to string.

## Review guidelines

Estimated Time of Review: 5 minutes

## Checklist before merging:

- [ ] Confirm that the relevant changelog(s) are up-to-date in case of any user-facing changes

### CODE REVIEW

File: source/state_representation/include/state_representation/space/cartesian/CartesianState.hpp [hunk 1]
```diff
@@ -598,4 +599,87 @@ inline void swap(CartesianState& state1, CartesianState& state2) {
...
+  } else if (case_insensitive_variable == "torque") {
+    return CartesianStateVariable::TORQUE;
+  } else if (case_insensitive_variable == "twist") {
+    return CartesianStateVariable::TWIST;
+  } else if (case_insensitive_variable == "wrench") {
+    return CartesianStateVariable::WRENCH;
+  } else if (case_insensitive_variable == "all") {
+    return CartesianStateVariable::ALL;
+  } else {
+    throw exceptions::InvalidStateException("Invalid Cartesian state variable: " + variable);
```
Comment (domire8) [inline source/state_representation/include/state_representation/space/cartesian/CartesianState.hpp:643]: I would prefer the more verbose option `InvalidStateVariableException` here, the `InvalidStateException` refers too much to the state itself IMO

**Review by domire8 — COMMENTED** [review]
Looks good, just one nitpick

File: source/state_representation/include/state_representation/space/cartesian/CartesianState.hpp (hunk 1 above)
Comment (bpapaspyros) [reply source/state_representation/include/state_representation/space/cartesian/CartesianState.hpp:643]: yes, fair enough, thanks

File: source/state_representation/include/state_representation/exceptions/InvalidStateVariableException.hpp [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <exception>
+#include <iostream>
+
+namespace state_representation::exceptions {
+class InvalidVariableStateException : public std::runtime_error {
```
Comment (domire8) [inline source/state_representation/include/state_representation/exceptions/InvalidStateVariableException.hpp:7]: Filename is good but not class name :smile:

Comment (bpapaspyros) [reply source/state_representation/include/state_representation/exceptions/InvalidStateVariableException.hpp]: yes yes, already fixed. Sorry

**Comment by domire8** [comment]
Okay first part is done, now you also need to update the python bindings. Let me know if you need help with this

File: python/source/state_representation/bind_exceptions.cpp [hunk 3]
```diff
@@ -11,6 +12,7 @@
 
 void bind_exceptions(py::module_& m) {
   py::register_exception<exceptions::EmptyStateException>(m, "EmptyStateError", PyExc_RuntimeError);
+  py::register_exception<exceptions::EmptyStateException>(m, "InvalidStateVariableException", PyExc_RuntimeError);
```
Comment (domire8) [inline python/source/state_representation/bind_exceptions.cpp:15]:
```suggestion
  py::register_exception<exceptions::InvalidStateVariableException>(m, "InvalidStateVariableError", PyExc_RuntimeError);
```

**Review by domire8 — COMMENTED** [review]
Nice

**Comment by bpapaspyros** [comment]
> Nice

I noticed we didn't make the move to C++20 in control-libraries. Should we do that (in general)? Here or separate PR?

**Comment by domire8** [comment]
> I noticed we didn't make the move to C++20 in control-libraries. Should we do that (in general)? Here or separate PR?

Not sure what that would imply. What would we gain from it? 

You could also open a backlog issue to do the equivalent changes in JointState

**Review by domire8 — APPROVED** [review]
Works for me

**Comment by bpapaspyros** [comment]
> > I noticed we didn't make the move to C++20 in control-libraries. Should we do that (in general)? Here or separate PR?
> 
> Not sure what that would imply. What would we gain from it?

at the moment nothing, it's not blocking. It's more for future proofing and to gain access to some more modern features that are always nice to have. (definitely not an immediate need)

**Pull request merged**